dashboard/app/controllers/api/v1/sections_controller.rb

```diff
@@ -175,7 +175,7 @@ def join
       return
     end
     render json: {
-      sections: current_user.sections_as_student.map(&:summarize_without_students),
+      sections: current_user.sections_as_student.reload.map(&:summarize_without_students),
       studentSections: current_user.sections_as_student_participant.map(&:summarize_without_students),
       plSections: current_user.sections_as_pl_participant.map(&:summarize_without_students),
```

Comment:
should we also reload `studentSections` and `plSections`?

Reply:
Both those methods filter the `sections_as_student` collection data:
https://github.com/code-dot-org/code-dot-org/blob/4ff077e255571bd39479f7671486a98e655b8692/dashboard/app/models/user.rb#L2014-L2020